gateway-service/src/main/java/org/zowe/apiml/gateway/conformance/VerificationOnboardService.java

```diff
@@ -43,18 +54,82 @@ public boolean checkOnboarding(String serviceId) {
 
 
     /**
-     * Accepts metadata and retrieves the Swagger url if it existsd
+     * Accepts metadata and retrieves the Swagger url if it exists
      *
      * @param metadata to grab swagger from
      * @return SwaggerUrl when able, empty string otherwise
      */
-    public String retrieveSwagger(Map<String, String> metadata) {
-        String swaggerUrl = metadata.get("apiml.apiInfo.api-v2.swaggerUrl");
+    public String findSwaggerUrl(Map<String, String> metadata) {
+
+        String swaggerKey = null;
+        for (String key : metadata.keySet()) {
+            if (key.contains("swaggerUrl")) {        // Find the correct key for swagger docs, can be both apiml.apiInfo.0.swaggerUrl or apiml.apiInfo.api-v1.swaggerUrl for example
+                swaggerKey = key;
+                break;
+            }
+        }
+        if (swaggerKey == null) {
+            return "";
+        }
+        String swaggerUrl = metadata.get(swaggerKey);
         if (swaggerUrl != null) {
             return swaggerUrl;
         }
         return "";
     }
 
 
+    /**
+     * Retrieves swagger from the url
+     *
+     * @param swaggerUrl URL to retrieve from
+     * @return Swagger as string
+     */
+    public String getSwagger(String swaggerUrl) {
+        String response;
+        response = restTemplate.getForEntity(swaggerUrl, String.class).getBody();
+        return response;
+    }
+
+    /**
+     * Checks if endpoints can be called and return documented responses
+     *
+     * @param getEndpoints GET endpoints to check
+     * @return List of problems
+     */
+    public List<String> testGetEndpoints(Set<Endpoint> getEndpoints) {
+        ArrayList<String> result = new ArrayList<>();
+        boolean gotResponseDifferentFrom404 = false;
+
+        for (Endpoint endpoint : getEndpoints) {
+            String urlFromSwagger = endpoint.getUrl();
+            // replaces parameters in {} in query
+            String url = urlFromSwagger.replaceAll("\\{[^{}]*}", "dummy");
+
+            ResponseEntity<String> response;
+            try {
+                response = restTemplate.getForEntity(url, String.class);
+            } catch (HttpClientErrorException | HttpServerErrorException e) {
+                response = ResponseEntity.status(e.getRawStatusCode()).headers(e.getResponseHeaders())
+                    .body(e.getResponseBodyAsString());
+            }
+
+            if (response.getStatusCode() != HttpStatus.NOT_FOUND) {
+                gotResponseDifferentFrom404 = true;
+            }
```

Comment:
Instead of ifs, with restTemplate you can also have different catch clauses for specific errors, something like:
```java
catch (HttpClientErrorException.NotFound e) {
    // just 404
} catch (HttpClientErrorException | HttpServerErrorException e) {
    // everything else
}
```

Reply:
Does not work in this particular case I think. I need to change the flag when I did**not** get a 404 response. This includes the case when there isn't an exception raised, so I don't think there is a way to avoid an if statement.